src/object/cli_obj.c

```diff
@@ -4886,10 +4912,7 @@ obj_ec_valid_shard_get(struct obj_auxi_args *obj_auxi, uint8_t *tgt_bitmap,
 	if (rc) {
 		if (force_ec_degrade) {
 			obj_auxi->no_retry = 1;
-			if (obj_auxi->csum_retry)
-				rc = -DER_CSUM;
-			else if (obj_auxi->tx_uncertain)
-				rc = -DER_TX_UNCERTAIN;
+			rc = retry_errcode(obj_auxi);
```

Comment:
logic changes a bit here, rc here might be overwrote with EIO incorrectly?

rc1 = retry_errorcode(obj_auxi);
if (rc1 != DER_IO)
  rc = rc1

Reply:
Yes, I'll fix it. thanks